cmake-edit-core/src/main/java/ru/urururu/cmakeedit/core/checker/SimulationState.java

```diff
@@ -12,19 +12,109 @@
     private final List<CommandInvocationNode> nodes;
     private int position;
     private final Map<String, Set<CommandInvocationNode>> variables;
-    private final Map<String, LogicalBlock> subroutines = new HashMap<>();
+    private final Set<CommandInvocationNode> suspiciousPoints;
+    private final Map<String, AbstractSimulator.CommandSimulator> dynamicSimulators = new HashMap<>();
 
-    SimulationState(List<CommandInvocationNode> nodes, int position) {
-        this(nodes, position, new HashMap<>());
+    SimulationState(List<CommandInvocationNode> nodes, int position, Set<CommandInvocationNode> suspiciousPoints) {
+        this(nodes, position, suspiciousPoints, new HashMap<>());
     }
 
-    SimulationState(List<CommandInvocationNode> nodes, int position, Map<String, Set<CommandInvocationNode>> variables) {
+    SimulationState(List<CommandInvocationNode> nodes, int position, Set<CommandInvocationNode> suspiciousPoints, Map<String, Set<CommandInvocationNode>> variables) {
         this.nodes = nodes;
         this.position = position;
         this.variables = variables;
+        this.suspiciousPoints = suspiciousPoints;
     }
 
-    void simulate(Set<Node> suspiciousPoints, Node node) {
+    String getNodeFromState(Node argumentNode) {
+        StringBuilder sb = new StringBuilder();
+
+        argumentNode.visitAll(new NodeVisitorAdapter() {
+            @Override
+            public void accept(ExpressionNode node) {
+                StringBuilder exprBuilder = new StringBuilder();
+
+                if (node.getKey() != null) {
+                    exprBuilder.append(node.getKey());
+                }
+
+                node.getNested().stream().forEach(n -> exprBuilder.append(getNodeFromState(n)));
+
+                String expr = exprBuilder.toString();
+                getValue(expr);
+
+                sb.append('*').append(expr);
+            }
+
+            @Override
+            public void accept(ConstantNode node) {
+                sb.append(node.getValue());
+            }
+        });
+
+        return sb.toString();
+    }
+
+    String getValue(ArgumentNode argumentNode) {
+        getNodeFromState(argumentNode);
+
+        return getValue(getArgument(argumentNode));
+    }
+
+    String getValue(String expression) {
+        int expressionStart = expression.indexOf("${"); // todo add support for ENV
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)